alpine/docker-entrypoint.sh

```diff
@@ -7,9 +7,17 @@ if [[ "$1" == "kong" ]]; then
   PREFIX=${KONG_PREFIX:=/usr/local/kong}
 
   if [[ "$2" == "docker-start" ]]; then
-    kong prepare -p "$PREFIX"
+    chown kong "$PREFIX"
+    su-exec kong kong prepare -p "$PREFIX"
+    
+    if [ ! -z ${KONG_STREAM_LISTEN+x} ]; then
```

Comment:
This check isn't correct. The transparent option gets used with http too.

Reply:
afraid I don't understand what you mean. TLDR what **should** the check and the command be keeping in mind alpine and centos compatibility?

Reply:
There should be no `if` here.

Reply:
There needs to be an if of some sort. We only need to setcap for transparent

Reply:
One condition could be if stream_listen, admin_listen or proxy_listen contain `transparent`. But we really just want it unconditional...

Reply:
I feel like we don't want it unless I'm unaware of some future plan we have

If I personally was running Kong I would want it's security profile restricted explicitly to what it requires and no more. So if I'm running with transparent then setcap is fine otherwise IMO it's an unnecessary privilege

Also the docker pr that lead to this suggested this route ...

Reply:
> One condition could be if stream_listen, admin_listen or proxy_listen contain transparent.

Can't add those conditions as well to make it more correct? @james-callahan

Reply:
We can, though the simplest implementation of that check (checking if they contain the substring `transparent`) may have false positives if it is found in e.g. the host.
I feel like it isn't a huge issue to have the capability and not use it.

Reply:
Maybe a regular expression to check if it’s specified after the host will do it?

Reply:
Yes that is possible; though you will need to repeat the check for each of the listen variables, and to make sure it handles the comma separated array form correctly.

Reply:
For your convenience this is regex that worked for me: `((?!transparent))\s+transparent`

It uses a lookahead to prevent matching any hostname that includes the word `transparent` in it, and only matches the actual `transparent` property after the hostname.

Reply:
You can play with the Regex here: https://www.regexpal.com/?fam=106839

Reply:
That fails on e.g. `localhost:3333, transparent:4444, bar:5555 ssl`

Reply:
How about this: `^([A-Za-z0-9:,\s\.]*\s+)transparent(\s+[\sA-Za-z0-9_]+)*$`

If it fully matches, then `transparent` is set, otherwise it's not: https://www.regexpal.com/?fam=106862

Reply:
This regex seems to be working for me: `^([A-Za-z0-9:,\s\.]+\s+)transparent((\s+[A-Za-z0-9_]*)|(\s*,[A-Za-z0-9:,\s\.]+))*$` ([test it here]( https://www.regexpal.com/?fam=106863)).

Reply:
This simplified version works and detects tricky use-cases as well: `[^\s,]+\s+transparent\b`

E.g.
* `host:2313 ssl transparent_not, host:8000 transparent_not, host:9000 transparent, host:7000 transparent`
* `   transparent, host:7000 transparent`

It is also able to determine how many instances of `transparent` there are, and extract their corresponding listener (although we don't care in this case).